### PR TITLE
Hadamard power added

### PR DESCRIPTION
**Pull request by Upabjojr**
#### Release Notes

* matrices
  * added support for Hadamard (i.e. elementwise) power of matrices.

### CODE REVIEW

**Review by sylee957 — APPROVED** [review]

**Comment by sylee957** [comment]
Can this be added under `__init__.py`, so that it can be imported with SymPy main module?

**Comment by Upabjojr** [comment]
@sylee957 thanks for your feedback. Do you have any suggestions for the Unicode/LaTeX printer?

I was thinking of printing `HadamardProduct(A, n)` as `A^{\circ n}` or, in unicode:
```
 ∘n
A
```
where the circle ∘ makes the distinction with the matrix power.

**Comment by Upabjojr** [comment]
The printers for `HadamardProduct` should be updated to include `⊘` or `\oslash` when followed by a negative `HadamardPower`.

File: sympy/matrices/expressions/hadamard.py [hunk 1]
```diff
@@ -81,3 +81,47 @@ def validate(*args):
...
+    def exp(self):
+        return self._args[1]
+
+    @property
+    def shape(self):
+        return self.base.shape
+
+    def _entry(self, i, j, **kwargs):
+        return self.base[i, j]**self.exp
+        return Pow(self.base[i, j], self.exp)
```
Comment (oscargus) [inline sympy/matrices/expressions/hadamard.py]: Double returns here.

**Comment by smichr** [comment]
- [ ] update printers (per above comment)

**Comment by Upabjojr** [comment]
>     * [ ]  update printers (per above comment)

I will open an issue after merging (have other things to focus on right now). Furthermore, taking some time to think of a proper printer is reasonable.

**Comment by Upabjojr** [comment]
If no one objects, I would merge this.

**Comment by oscarbenjamin** [comment]
> I will open an issue after merging (have other things to focus on right now). Furthermore, taking some time to think of a proper printer is reasonable.

That makes sense but why the rush to merge? If you need time then take time.

**Comment by sylee957** [comment]
Adding printers for latex, mathml, pprint, or such is trivial.
This can be done separately by anyone who can take up the issue after the merge, but whatever can be fine if you have time to merge some PR to this branch.

**Comment by Upabjojr** [comment]
I want to focus on matrix derivatives right now. Derivatives of Hadamard expressions should be supported.

**Comment by oscargus** [comment]
A reminder here that `str` should give a copy-and-pastable expression. See #17138

**Comment by Upabjojr** [comment]
> A reminder here that `str` should give a copy-and-pastable expression. See #17138

Well, maybe `srepr` or `repr` should. I'm fine with `str` having some (Python-invalid) shortcuts in order to make the output readable.

**Pull request merged**